File: MusicVue/src/types/music.ts
export interface Music {
  id: number
  title: string
  artist: string
  album?: string
  fileUrl: string
  previewUrl?: string
  duration?: number
  genre?: string
  lyrics?: string  // 歌词内容(LRC格式)
  coverUrl?: string
  playCount?: number
  likeCount?: number
  status?: number
}

export interface RecommendRecord {
  id: number
  uploadId: number
  musicId: number
  similarity: number
  rank: number
  music?: Music
  createTime?: string
}

export interface UploadRecord {
  id: number
  userId: number
  fileUrl: string
  fileName: string
  fileSize: number
  fileType?: string
  status: 'UPLOADING' | 'ANALYZING' | 'COMPLETED' | 'FAILED'
  features?: string
  analysisResult?: string
  errorMessage?: string
  createTime?: string
}

export interface AnalysisResult {
  uploadId: number
  features: number[]
  analysisResult: string
  recommendations: RecommendRecord[]
}

File: src/main/java/com/example/personalmusicsystem/entity/Music.java
package com.example.personalmusicsystem.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 音乐实体类
 */
@Data
public class Music {
    private Long id;
    private String title;
    private String artist;
    private String album;
    private String fileUrl;
    private String previewUrl;
    private Integer duration;
    private String genre;
    private String lyrics;  // 歌词内容(LRC格式)
    private String features;  // JSON字符串，MFCC等特征向量
    private String coverUrl;
    private Integer playCount;
    private Integer likeCount;
    private Integer status;  // 1-正常 0-下架
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

